File: x-fly-http/src/main/java/com/xia/fly/http/request/GetRequest.java
package com.xia.fly.http.request;

import android.support.annotation.NonNull;

import io.reactivex.Observable;
import okhttp3.ResponseBody;

/**
 * @author xia
 * @date 2018/8/3.
 */
public class GetRequest extends BaseRequest<GetRequest> {
    private boolean mIsDownload;

    public GetRequest(@NonNull String url) {
        super(url);
    }

    public GetRequest upDownload() {
        mIsDownload = true;
        return this;
    }

    @Override
    protected Observable<ResponseBody> generateRequest() {
        if (mIsDownload) {
            return mRestService.downloadFile(mUrl);
        }
        return mRestService.get(mUrl, mParams);
    }
}

File: fly-http/src/main/java/com/xia/fly/http/request/GetRequest.kt
package com.xia.fly.http.request

import io.reactivex.Observable
import okhttp3.ResponseBody

/**
 * @author xia
 * @date 2018/8/3.
 */
@Suppress("unused")
class GetRequest(url: String) : BaseRequest<GetRequest>(url) {
    private var mIsDownload: Boolean = false

    fun upDownload(): GetRequest {
        mIsDownload = true
        return this
    }

    override fun generateRequest(): Observable<ResponseBody>? {
        return if (mIsDownload) {
            mRestService?.downloadFile(mUrl, mHeaders)
        } else mRestService?.get(mUrl, mParams, mHeaders)
    }
}
